src/main/java/org/folio/inventory/resources/AbstractInstances.java

```diff
@@ -118,10 +126,41 @@ private CompletableFuture<List<Response>> updateInstanceRelationships(Instance i
       });
     }
 
-    List<CompletableFuture<Response>> allFutures = update(instanceRelationshipsClient,
-      existingRelationships, updatingRelationships);
+    return validateNewRelationships(context, existingRelationships, updatingRelationships, instance)
+      .thenCompose(v -> allResultsOf(update(instanceRelationshipsClient, existingRelationships, updatingRelationships)));
+  }
 
-    return allResultsOf(allFutures);
+  private CompletableFuture<List<Instance>> validateNewRelationships(WebContext context, Map<String, InstanceRelationship> existingRelationships,
+                                                                     Map<String, InstanceRelationship> updatingRelationships, Instance instance) {
+    return consortiumService.getConsortiumConfiguration(context).toCompletionStage().toCompletableFuture()
+      .thenCompose(consortiumConfigurationOptional -> {
+        List<CompletableFuture<Instance>> validateNewRelationshipFutures = new ArrayList<>();
+        if (consortiumConfigurationOptional.isPresent()) {
+          List<InstanceRelationship> newRelationships = updatingRelationships.keySet().stream().filter(key -> !existingRelationships.containsKey(key))
+            .map(updatingRelationships::get).toList();
+
+          newRelationships.forEach(relationship -> {
+            String instanceId = getRelatedInstanceId(instance, relationship);
+            validateNewRelationshipFutures.add(checkIfInstanceExistAtLocalMember(context, instanceId));
+          });
+        }
+        return allResultsOf(validateNewRelationshipFutures);
+      });
+  }
+
+  private CompletableFuture<Instance> checkIfInstanceExistAtLocalMember(WebContext context, String instanceId) {
+    return InstanceUtil.findInstanceById(instanceId, storage.getInstanceCollection(context))
+      .recover(throwable -> {
+        if (throwable instanceof NotFoundException) {
+          log.warn("confirmInstanceExistAtLocalMember:: Trying to link shared and local instance, instanceId: {}", instanceId);
+          return Future.failedFuture(new BadRequestException(LINKING_SHARED_AND_LOCAL_INSTANCE_ERROR));
+        }
+        return Future.failedFuture(throwable);
+      }).toCompletionStage().toCompletableFuture();
+  }
+
+  private static String getRelatedInstanceId(Instance instance, InstanceRelationship relationship) {
+    return !relationship.superInstanceId.equals(instance.getId()) ? relationship.superInstanceId : relationship.subInstanceId;
```

Comment:
Can superInstanceId can be null here? If yes, there should be added a check here to avoid NPE.

Reply:
It cannot be null, thanks